### PR TITLE
Deploy Stage

### PR DESCRIPTION
**Pull request by nlordell**
Refactor Travis file to include a deploy stage so that the build does not try to deploy on every job.

### Test Plan

0.2.0 auto deployment.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -13,25 +18,40 @@ env:
...
+      env:
+        - RUSTC_WRAPPER=
+      before_install: skip
+      before_script: skip
+      script: skip
+      deploy:
+        provider: script
+        script: bash ci/deploy.sh
+        on:
+          tags: true
```
Comment (fleupold) [inline .travis.yml:49]: I'm confused. Why is this required if the stage is only run when a tag is set?

File: .travis.yml [hunk 2]
```diff
@@ -13,25 +18,40 @@ env:
...
         - cargo run --example linked
         - cargo run --package examples-generate
         - cargo run --example rinkeby
 
+    - name: "crates.io"
+      stage: "Deploy"
+      language: rust
+      rust: stable
+      env:
+        - RUSTC_WRAPPER=
```
Comment (fleupold) [inline .travis.yml:41]: Why no caching here?

File: .travis.yml (hunk 1 above)
Comment (nlordell) [reply .travis.yml:49]: Because or travis needs a condition to deploy on, I had it off and got this:
```
Skipping a deployment with the script provider because this branch is not permitted: deploy-test
```
https://travis-ci.org/gnosis/ethcontract-rs/jobs/627650549#L243

File: .travis.yml (hunk 2 above)
Comment (nlordell) [reply .travis.yml:41]: More of a just in case - I am paranoid. Also I figured deployment happens rarely enough that cache speedup isn't worth it.

Comment (nlordell) [reply .travis.yml:41]: Oh, a little background information - `cargo publish` does a full build before publishing. I am also unsure about `RUSTC_WRAPPER` interaction with `cargo publish` since it seams to completely rebuild regardless of what is already in `target/`

File: .travis.yml [hunk 3]
```diff
@@ -1,5 +1,10 @@
 if: (branch = master) OR (type = pull_request) OR (tag IS present)
+stages:
+  - name: "Build and Test"
+  - name: "Deploy"
+    if: tag IS present
 
+stage: "Build and Test"
```
Comment (fleupold) [inline .travis.yml:7]: Do we need this stage name twice? It's also in line 24.

**Review by fleupold — APPROVED** [review]

File: .travis.yml (hunk 3 above)
Comment (nlordell) [reply .travis.yml:7]: Don't see why we would, giving it a try.
